src/fastfield/reader.rs

```diff
@@ -61,73 +65,100 @@ pub trait FastFieldReader<Item: FastValue>: Clone {
     pub fn open(file: FileSlice) -> crate::Result<DynamicFastFieldReader<Item>> {
-        Ok(DynamicFastFieldReader::Bitpacked(
-            BitpackedFastFieldReader::open(file)?,
-        ))
+        let bytes = file.read_bytes()?;
+        let (mut id_bytes, data_bytes) = bytes.split(1);
+        let id = u8::deserialize(&mut id_bytes)?;
+
+        let reader = match id {
+            BitpackedFastFieldSerializer::<Vec<u8>>::ID => {
+                DynamicFastFieldReader::Bitpacked(FastFieldReaderCodecWrapper::<
+                    Item,
+                    BitpackedReader,
+                >::open_from_bytes(data_bytes)?)
+            }
+            LinearInterpolFastFieldSerializer::ID => {
+                DynamicFastFieldReader::LinearInterpol(FastFieldReaderCodecWrapper::<
+                    Item,
+                    LinearinterpolFastFieldReader,
+                >::open_from_bytes(
+                    data_bytes
+                )?)
+            }
+            _ => {
+                panic!(
+                    "unknown fastfield id {:?}. Data corrupted or using old tantivy version.",
+                    id
+                )
+            }
+        };
+        Ok(reader)
     }
 }
 
 impl<Item: FastValue> FastFieldReader<Item> for DynamicFastFieldReader<Item> {
     fn get(&self, doc: DocId) -> Item {
         match self {
             Self::Bitpacked(reader) => reader.get(doc),
+            Self::LinearInterpol(reader) => reader.get(doc),
         }
     }
     fn get_range(&self, start: DocId, output: &mut [Item]) {
         match self {
             Self::Bitpacked(reader) => reader.get_range(start, output),
+            Self::LinearInterpol(reader) => reader.get_range(start, output),
         }
     }
     fn min_value(&self) -> Item {
         match self {
             Self::Bitpacked(reader) => reader.min_value(),
+            Self::LinearInterpol(reader) => reader.min_value(),
         }
     }
     fn max_value(&self) -> Item {
         match self {
             Self::Bitpacked(reader) => reader.max_value(),
+            Self::LinearInterpol(reader) => reader.max_value(),
         }
     }
 }
 
-/// Trait for accessing a fastfield.
+/// Wrapper for accessing a fastfield.
+///
+/// Holds the data and the codec to the read the data.
 ///
-/// Depending on the field type, a different
-/// fast field is required.
 #[derive(Clone)]
-pub struct BitpackedFastFieldReader<Item: FastValue> {
+pub struct FastFieldReaderCodecWrapper<Item: FastValue, CodecReader> {
+    reader: CodecReader,
     bytes: OwnedBytes,
-    bit_unpacker: BitUnpacker,
-    min_value_u64: u64,
-    max_value_u64: u64,
     _phantom: PhantomData<Item>,
 }
 
-impl<Item: FastValue> BitpackedFastFieldReader<Item> {
+impl<Item: FastValue, C: CodecReader> FastFieldReaderCodecWrapper<Item, C> {
     /// Opens a fast field given a file.
     pub fn open(file: FileSlice) -> crate::Result<Self> {
```

Comment:
When is this function useful? Is it for when we are sure of the encoding and we want static dispatch or something?

Reply:
there are some cases, where we always use bitpacked for now and there we can use static dispatch